### PR TITLE
[dhctl] Waiting for cluster become bootstrapped state and output diagnostic messages about cloud ephemeral nodes

### LINKED ISSUE #2234: [dhctl] Check node creating for CloudEphemeral node-groups (from resources.yaml)
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

Very often we are using `resources.yaml` for creating node-groups and bootstrap nodes for them (system, frontend, worker).
And very often we and users have problem with bootstrapping nodes for node-groups like [this](https://t.me/deckhouse_ru/7027) or ingress-controller resource was created but frontend nodes can not be created. Often users set wrong flavor for instance or have quotas in a cloud.

It was confused.  Users should find masters IP's login to it and find problems (with `kubectl -n default get event --sort-by='metadata.creationTimestamp'`). Or bootstrap finish successfully  but users can not use cluster because frontend can not be created. I think it is wrong.

### Proposed Solution

Because `NodeGroup` is Deckhouse resource we can filter them in `resources.yaml` and wait until ready nodes in node group less than minimal nodes in node group. If node group has error while creating nodes (we can get last error in the status) display them immediately.

### Additional Information

Log with error timeout (write for fast find)

```
Resources to create:
│ deckhouse.io/v1, Kind=IngressNginxController
│ deckhouse.io/v1, Kind=ClusterAuthorizationRule
Bootstrap: Create Resources (900.00 seconds) FAILED
```

### PR DESCRIPTION
**Pull request by name212**
## Description

Now we are waiting for create config map  `kube-system/d8-cluster-is-bootstraped`. When we are waiting creation dhctl output information about nodegroups and print machine failed creation events. 

Example output:
image

Happy case:
image

## Why do we need it, and what problem does it solve?

Closes #2234

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: dhctl
type: feature
summary: Waiting for cluster become bootstrapped state and output diagnostic messages about cloud ephemeral nodes
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: dhctl/pkg/kubernetes/actions/resources/cluster_is_bootstrap.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+func (n *clusterIsBootstrapCheck) IsReady() (bool, error) {
+	defer func() {
+		n.attempts++
+		n.logger.LogInfoF("\n")
+	}()
+
+	n.logger.LogInfoF("Waiting for cluster will be in 'bootstrapped' state:\n")
+
+	notBootstrappedMsg := "Cluster is not yet bootstrapped. Waiting for at least one non-master node in Ready status.\n"
```
Comment (nabokihms) [inline dhctl/pkg/kubernetes/actions/resources/cluster_is_bootstrap.go:229]:
```suggestion
	notBootstrappedMsg := "The cluster has not been bootstrapped yet. Waiting for at least one non-master node in Ready status.\n"
```

File: dhctl/pkg/kubernetes/actions/resources/cluster_is_bootstrap.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+	ok, err := n.hasBootstrappedCM()
+	if err != nil {
+		n.logger.LogDebugF("Error while checking cluster state: %v\n", err)
+		n.logger.LogInfoF(notBootstrappedMsg)
+		return false, nil
+	}
+
+	if ok {
+		n.logger.LogInfoF("Cluster is bootstrapped. Waiting for resource creation.\n")
```
Comment (nabokihms) [inline dhctl/pkg/kubernetes/actions/resources/cluster_is_bootstrap.go:239]:
```suggestion
		n.logger.LogInfoF("The cluster is bootstrapped. Waiting for the creation of resources.\n")
```

**Review by nabokihms — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/openstack'...

**Pull request merged**